### PR TITLE
Add isComplex and refactor ProjectData getters

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Add isComplex public function and private member variable
Move inline functions to a single translation unit
Add test
Related to #4099 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature
- Refactoring (no functional changes, no api changes)
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 22.04, must pass CI

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Review by ye-luo — APPROVED** [review]

File: src/Utilities/ProjectData.h [hunk 1]
```diff
@@ -134,6 +142,9 @@ class ProjectData
 
   // The driver version of the project
   DriverVersion driver_version_;
+
+  // tracks if complex wavefunction is used, true: complex, false: real
+  bool is_complex_;
```
Comment (ye-luo) [inline src/Utilities/ProjectData.h:147]: This needs to be initialized. Probably for now based on QMC_COMPLEX

**Review by ye-luo — CHANGES_REQUESTED** [review]

File: src/Utilities/ProjectData.h (hunk 1 above)
Comment (ye-luo) [reply src/Utilities/ProjectData.h:147]: Please do a rebase once this is fixed.

Comment (williamfgc) [reply src/Utilities/ProjectData.h:147]: Thanks @ye-luo , that was a complete oversight from my side. Fixed it.

Comment (ye-luo) [reply src/Utilities/ProjectData.h:147]: Our CI actually reported failure. So it well served as a test for CI.

Comment (williamfgc) [reply src/Utilities/ProjectData.h:147]: Interesting, I am using vanilla Ubuntu 22 and the local unit tests passed. The gcc compiler might be too permissive.

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Review by PDoakORNL — APPROVED** [review]
LGTM

**Pull request merged**